Add AI Investigations section with Single-View AI Investigation page (#36350)
DataDog/documentation: content/en/real_user_monitoring/ai_investigations/single_view_ai_investigation.md

---
title: Single-View AI Investigation
description: "Run an agentic investigation on a single RUM view to surface root causes of poor user experience."
further_reading:
  - link: "/real_user_monitoring/ai_investigations/"
    tag: "Documentation"
    text: "AI Investigations overview"
  - link: "/real_user_monitoring/explorer/"
    tag: "Documentation"
    text: "RUM Explorer"
  - link: "/real_user_monitoring/explorer/events/"
    tag: "Documentation"
    text: "View event side panel"
---

## Overview

> Single-View AI Investigation runs an agentic root-cause analysis on a single RUM view. Datadog's RUM agent inspects the view event and its sub-events to surface structured symptoms and the most plausible explanations for why the user's experience was poor — long tasks blocking the main thread, slow backend responses, expensive third-party scripts, and so on.

Reviewer: ```suggestion
Single-View AI Investigation runs an agentic root-cause analysis on a single RUM view. When you find a session with poor performance, such as a page that loaded slowly or threw errors, click **Investigate**. Datadog's RUM agent inspects all the data attached to that view: Javascript errors, slow network requests, long tasks blocking the main thread, backend traces, CPU profiles, and device context. 
```

Author: Applied, thanks!

> The goal is to compress the path from signal to plausible explanation. Instead of manually parsing attributes and sub-events to understand what happened, you get a curated set of findings — ranked by impact — directly in the view side panel where you already work. From there, you can iterate through a chat interface and save the result to a [Notebook][1] for sharing.

Reviewer: ```suggestion
Instead of manually combing through RUM events to figure out whether the cause was a slow API call, a Javascript bundle, or a CDN issue, you get a ranked list of findings grouped by root-cause category: App Performance, Server Side, Third Party, and Environment. From there, you can follow up through a chat interface or save the results to a [Notebook][1] to share with your team.
```

Author: Applied, thanks!

{{< img src="real_user_monitoring/ai_investigations/single-view-ai-investigation-overview.png" alt="The Single-View AI Investigation panel showing categorized findings for a RUM view." style="width:100%;" >}}

## Data analyzed

> To investigate a view, Datadog's RUM agent inspects the data Datadog has collected for that view, and reaches out to correlated telemetry when it's available:

Reviewer: ```suggestion
To investigate a view, Datadog's RUM agent inspects the data Datadog has collected for that view and accesses correlated telemetry when it's available:
```

Author: Applied, thanks!

- **Browser profiling data**, when [Browser profiling][8] is enabled for the application. The agent uses CPU profiles to attribute App Performance findings to specific JavaScript functions in your code.

The richer the data available for the view, the more precise the analysis. Correlating RUM with APM and enabling Browser profiling helps the agent investigate beyond the frontend timeline.

## Launch an investigation

1. Open a RUM view side panel.
2. Click **Investigate**.

The investigation runs and streams results into the side panel as they become available, so you can start reading the first findings before the analysis is complete.

## Sources of root causes

Datadog's RUM agent looks at four sources to identify the root causes of poor performance on a view:

| Source            | What is examined                                                                                              |
|-------------------|---------------------------------------------------------------------------------------------------------------|

> | App Performance   | Browser and client-side issues, such as long tasks, JavaScript execution, and rendering delays.               |

Reviewer: ```suggestion
| App Performance   | Client-side issues, such as long tasks, JavaScript execution, and rendering delays.               |
```

Author: Applied, thanks!

| Server Side       | Backend latency and server-side errors that affected the view.                                                |
| Third Party       | Performance impact from third-party scripts and libraries loaded on the page.                                 |
| Environment       | Network and infrastructure conditions that affected the user's experience.                                    |

## Read the results

> Each finding is shown as a card with a title, a description of the issue, a severity level, and links to the affected events. When multiple findings are surfaced, they are ranked so you can focus on the highest-impact issues first.

Reviewer: ```suggestion
Each finding is shown as a card with a title, a description of the issue, a severity level, and links to the affected events. Multiple findings appear ranked by impact so you can focus on the highest-impact issues first.
```

Author: Applied, thanks!

> {{< img src="real_user_monitoring/ai_investigations/single-view-ai-investigation-results.png" alt="Results panel showing ranked findings with severity, descriptions, and links to the affected events." style="width:100%;" >}}

Reviewer: reminder to add image

Author: Image added in latest commit.

A lightweight chat interface lets you follow up on the analysis: ask for more detail on a specific finding, request additional context, or explore related symptoms.

> Investigations can be saved to a [Notebook][1] for sharing with your team or for future reference.

Reviewer: ```suggestion
Save investigations to a [Notebook][1] to share with your team for future reference.
```

Author: Applied, thanks!

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /notebooks/
[2]: /real_user_monitoring/application_monitoring/browser/monitoring_resource_performance/

> [3]: /real_user_monitoring/application_monitoring/browser/data_collected/#view-timing-attributes

Reviewer: ```suggestion
[3]: /real_user_monitoring/application_monitoring/browser/data_collected/#long-task-timing-attributes
```

Author: Applied, thanks!
